# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/gtest/TestDeviceInputTrack.cpp
# Commit: f81e1b1615b3
# Full Hash: f81e1b1615b3a9c5ef6c62a291f79029b3807dde
# Author: Karl Tomlinson <karlt+@karlt.net>
# Date: 2023-03-31 09:51:00
# Regressor Bug: 1801190
# File Overlap Count: 1
# Description:
#   Bug 1801190 do not reset mAudioStreamState to None on state changes from an old cubeb_stream r=pehrsons
#   
#   Such resets were preventing transitions to Running for new streams and allowing
#   MaybeStartAudioStream() to start another stream.
#   
# ==============================================================================

diff -r 9592d9865f5a -r f81e1b1615b3 dom/media/gtest/TestDeviceInputTrack.cpp
--- a/dom/media/gtest/TestDeviceInputTrack.cpp	Fri Mar 31 02:23:14 2023 +0300
+++ b/dom/media/gtest/TestDeviceInputTrack.cpp	Thu Mar 30 23:18:44 2023 +0000
@@ -263,7 +263,8 @@
                     sourceId, AudioInputSource::EventListener::State::Started));
     EXPECT_CALL(*listener,
                 AudioStateCallback(
-                    sourceId, AudioInputSource::EventListener::State::Stopped));
+                    sourceId, AudioInputSource::EventListener::State::Stopped))
+        .Times(2);
 
     // No input channels and device preference before start.
     EXPECT_EQ(track->NumberOfChannels(), 0U);
@@ -308,7 +309,8 @@
                     sourceId, AudioInputSource::EventListener::State::Started));
     EXPECT_CALL(*listener,
                 AudioStateCallback(
-                    sourceId, AudioInputSource::EventListener::State::Stopped));
+                    sourceId, AudioInputSource::EventListener::State::Stopped))
+        .Times(2);
 
     DispatchFunction([&] {
       track->StartAudio(MakeRefPtr<AudioInputSource>(
@@ -387,7 +389,8 @@
                   sourceId, AudioInputSource::EventListener::State::Started));
   EXPECT_CALL(*listener,
               AudioStateCallback(
-                  sourceId, AudioInputSource::EventListener::State::Stopped));
+                  sourceId, AudioInputSource::EventListener::State::Stopped))
+      .Times(2);
 
   DispatchFunction([&] {
     track->StartAudio(MakeRefPtr<AudioInputSource>(
@@ -462,7 +465,8 @@
                   sourceId, AudioInputSource::EventListener::State::Started));
   EXPECT_CALL(*listener,
               AudioStateCallback(
-                  sourceId, AudioInputSource::EventListener::State::Stopped));
+                  sourceId, AudioInputSource::EventListener::State::Stopped))
+      .Times(2);
 
   // Launch and start an audio stream.
   DispatchFunction([&] {
@@ -520,6 +524,9 @@
   EXPECT_CALL(*listener,
               AudioStateCallback(
                   sourceId, AudioInputSource::EventListener::State::Error));
+  EXPECT_CALL(*listener,
+              AudioStateCallback(
+                  sourceId, AudioInputSource::EventListener::State::Stopped));
 
   // Launch and start an audio stream.
   DispatchFunction([&] {
